src/services/signer/signer-modal-handler.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable import/no-unused-modules */
+/* eslint-disable import/prefer-default-export */
+import { EncodeObject, StdFee } from '@cybercongress/cyber-ts';
+import type { SignerModalRef } from 'src/components/signer-modal/signer-modal';
+
+interface SignerModalData {
+  resolve?: (value: unknown) => void;
+  reject?: (reason?: any) => void;
+  signerAddress?: string;
+  messages?: EncodeObject[];
+  fee?: number | 'auto' | StdFee;
+  memo?: string | undefined;
+}
+
+export class SignerModalHandler {
```

Comment:
maybe it can be context better? to use OOP less with react

Reply:
if it would be context it would be also redux (what's the difference to be honest?!)
it's done that way to make it possible to use it outside the react scope (literally in **CybSignerClient**)